src/components/Button/Button.tsx

```diff
@@ -188,20 +192,23 @@ const BaseButton = styled('button')<ButtonProps>(
  * The Button component enables the user to perform an action or navigate
  * to a different screen.
  */
-export function Button(props: BaseProps & LinkElProps): ReturnType;
-export function Button(props: BaseProps & ButtonElProps): ReturnType;
-export function Button({
-  children,
-  icon: Icon,
-  ...props
-}: ButtonProps): ReturnType {
-  const { Link } = useComponents();
-  const LinkButton = BaseButton.withComponent(Link);
-  const ButtonElement = props.href ? LinkButton : BaseButton;
-  return (
-    <ButtonElement {...props}>
-      {Icon && <Icon css={iconStyles} role="presentation" />}
-      {children}
-    </ButtonElement>
-  );
-}
+// export function Button(props: BaseProps & LinkElProps): ReturnType;
+// export function Button(props: BaseProps & ButtonElProps): ReturnType;
```

Comment:
These overloads are no longer necessary, right? 

I checked out the branch locally and it seems that the component is correctly typed.

Reply:
Actually it was a left over, I started like this and forgot to adjust them, now they are back